### DISCUSSION TITLE
Welcome to CMIYC Ideas!

### DISCUSSION THREAD

**Discussion by katemalash** [IT-Academy-Social-Projects-KRV/CMIYC#14] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ajderka** [comment]
### What technology we use:
- Java 11
- loombook
- JUnit tests
- MySQL
- Mokito
- jackson
- Docker
- postman
- webstorm (studio)
- Spring Web
- MSC Generator

**Comment by Ajderka** [comment]
### **Highlight**

> TOSEE - Docker hub 
> TODO - Spring Date REST (try to rise project) - We decided, that we use another technology. We use Spring Boot Starter Web.
> TODO - Write docker file
> TODO - write about scheme auth > how many roles we must use in our application

**Skeleton technologies (dependecies):** 
- Spring Boot Starter Web
- Loom Book
- Spring Boot Starter Test
- Spring Boot Dev Tools

[For Angular learning](https://angular.io/tutorial)

**We have to decide how our microservices will connect to each other, by what technology :**
- [RestTemplate](https://www.baeldung.com/rest-template)
- [Feign](https://www.baeldung.com/intro-to-feign)

### Authorization topic
1. Admin (manage schemas) - add file schema.cfg (validate - xml, size <10mb, CRD)
2. Admin (manage users) -list of users CRUD edit without email users, manage Admin
3. User - can use some common operation SCHEMA (правила создания запросов к API)

**Comment by ZX-pool** [comment]
22.02.2022
Ideas!
1. Create intern speaking club!
2. Speaking about usefull questions in interviews
3. Optimize work with GIT, who approved pull requests
4. Create subdirectories for each other of directions, corresponding components in same folder
5. Always write status before daily meeting in github discussion in Daily SCRUM section
6. Make usefull structure schema of project
7. Create readme for setup database parameters

**Comment by ZX-pool** [comment]
24.03.2022
Idea!
1. Save responses in database
2.  Cron job for deleting old responses (10-14 days)
3. One Request_ID can have up to 3 Response_ID

**Comment by ZX-pool** [comment]
30.03.22
Ideas after second sprint:
1. Better description of tasks
2. Better work with GIT  - create instruction (cards not always closed when they are done, not all branches named as we talk, not all pull request have information like assignee, linked task, label,describe tasks in more detail, do not fix the errors of one feature in another, create a coherent instruction for such actions)
3. Url mappings convention
4. Think and write 3-4 tasks for backlog before sprint planning, and we will have much more ideas and it will be faster
5. At the start of the demo - make a full project flow,describe all things so a person which not understand the project can easily figure out what we are doing.

**Comment by JohnnyMahoney** [comment]
### 12 april 22
Ideas after third sprint:

**Reply by JohnnyMahoney** [reply]
If you are not able to do the task,please,tell it to the team
Don't ignore our daily meetings message,write it everyday
Spend less time on not working issues(add task not only on planning,also on the sprint)
Discuss our code convention
More attention on pull requests(take a look at the code convention)
gather together on discord)))
fix old bugs
more attention on tests!
If you haven't an opportunity to take a part on Demo at the end of the sprint,give your task to someone to show it on Demo

**Comment by Gorbokonyk** [comment]
![image](https://user-images.githubusercontent.com/19592315/165750490-24f6a074-e624-4a75-af89-93b26b39705f.png)
